src/paths.md

```diff
@@ -62,7 +62,10 @@ mod m {
 > &nbsp;&nbsp; | [_SimplePathSegment_]
 >
 > _GenericArgsBinding_ :\
-> &nbsp;&nbsp; [IDENTIFIER] `=` [_Type_]
+> &nbsp;&nbsp; [IDENTIFIER] _GenericArgs_<sup>?</sup> `=` [_TypeParamBounds_]
```

Comment:
I'm a little confused by this change to GenericArgsBinding. Why is it changing from `=Type` to `=TypeParamBounds`?

From what I can tell, this is in [`parse_assoc_equality_term`](https://github.com/rust-lang/rust/blob/42825768b103c28b10ce0407749acb21d32abeec/compiler/rustc_parse/src/parser/path.rs#L668-L676), which looks like it is expecting a type.

Can you walk me through what I'm missing? For example, `x::<foo<X>=impl Trait + ?Foo>`. The `impl Trait` part is the where I'm getting lost. If the `=` part is supposed to be a `TypeParamBounds`, how does that parse? If it is `TypeParamBounds`, then it looks like that follows to `TraitBound` which is essentially a `TypePath`, and `TypePath` does not include `Type` (it is mostly just paths with identifiers).

Reply:
This is a copy/paste error, lol

Reply:
This should definitely just be `= TYPE`